csrc/polymorphic_value.h

```diff
@@ -181,6 +181,12 @@ class StructHandle {
   StructHandle& operator=(const StructHandle& other) = default;
   StructHandle& operator=(StructHandle&& other) = default;
 
+  //! This is a shallow comparison operator that just checks whether we point to
+  //! the same exact Struct
+  bool operator==(const StructHandle& other) const {
+    return struct_ptr_ == other.struct_ptr_;
+  }
+
```

Comment:
After binding a `TensorMetaData` handle in `PolymorphicValues`, without this I was hitting the following error:
>    what():  ret.has_value() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/lib/dynamic_type/src/dynamic_type/dynamic_type.h":741, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Cannot compute N7nvfuser12StructHandleE == N7nvfuser12StructHandleE : incompatible type
Exception raised from operator== at /opt/pytorch/nvfuser/lib/dynamic_type/src/dynamic_type/dynamic_type.h:741 (most recent call first):

Adding this shallow pointer comparison fixes it, but @zasdfgbnm please let me know if this is safe to add.

Reply:
I think this makes sense